pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -548,11 +648,90 @@ func TestRetryPendingResizes(t *testing.T) {
+			expectedAllocatedPodReqs:     v1.ResourceList{v1.ResourceCPU: cpu1200m, v1.ResourceMemory: mem1200M},
+			expectedResize: []*v1.PodCondition{
+				{
+					Type:    v1.PodResizeInProgress,
+					Status:  "True",
+					Message: "",
+				},
+			},
+			expectPodSyncTriggered: "true",
+		},
+		{
+			name:                         "pod-level: Request CPU increase beyond current capacity - expect Deferred",
+			inPlacePodLevelResizeEnabled: true,
+			originalPodRequests:          v1.ResourceList{v1.ResourceCPU: cpu1500m, v1.ResourceMemory: mem1500M},
+			newPodRequests:               v1.ResourceList{v1.ResourceCPU: cpu2500m, v1.ResourceMemory: mem1500M},
+			originalRequests:             v1.ResourceList{v1.ResourceCPU: cpu1000m, v1.ResourceMemory: mem1000M},
+			newRequests:                  v1.ResourceList{v1.ResourceCPU: cpu1000m, v1.ResourceMemory: mem1000M},
+			expectedAllocatedReqs:        v1.ResourceList{v1.ResourceCPU: cpu1000m, v1.ResourceMemory: mem1000M},
+			expectedAllocatedPodReqs:     v1.ResourceList{v1.ResourceCPU: cpu1500m, v1.ResourceMemory: mem1500M},
+			expectedResize: []*v1.PodCondition{
+				{
+					Type:    v1.PodResizePending,
+					Status:  "True",
+					Reason:  "Deferred",
+					Message: "Node didn't have enough resource: cpu",
+				},
+			},
+			expectPodSyncTriggered: "true",
+		},
+		{
+			name:                         "pod-level: Request memory increase beyond node capacity - expect Infeasible",
+			inPlacePodLevelResizeEnabled: true,
+			originalPodRequests:          v1.ResourceList{v1.ResourceCPU: cpu1500m, v1.ResourceMemory: mem1500M},
+			newPodRequests:               v1.ResourceList{v1.ResourceCPU: cpu1500m, v1.ResourceMemory: mem4500M},
+			originalRequests:             v1.ResourceList{v1.ResourceCPU: cpu1000m, v1.ResourceMemory: mem1000M},
+			newRequests:                  v1.ResourceList{v1.ResourceCPU: cpu1000m, v1.ResourceMemory: mem1000M},
+			expectedAllocatedReqs:        v1.ResourceList{v1.ResourceCPU: cpu1000m, v1.ResourceMemory: mem1000M},
+			expectedAllocatedPodReqs:     v1.ResourceList{v1.ResourceCPU: cpu1500m, v1.ResourceMemory: mem1500M},
+			expectedResize: []*v1.PodCondition{
+				{
+					Type:    v1.PodResizePending,
+					Status:  "True",
+					Reason:  "Infeasible",
+					Message: "Node didn't have enough capacity: memory, requested: 4718592000, capacity: 4294967296",
+				},
+			},
+			expectPodSyncTriggered: "true",
+		},
 	}
 
 	for _, tt := range tests {
 		for _, isSidecarContainer := range []bool{false, true} {
+			if tt.inPlacePodLevelResizeEnabled && isSidecarContainer {
+				continue // pod level resources makes the distinction between container types irrelevant
+			}
 			t.Run(fmt.Sprintf("%s/sidecar=%t", tt.name, isSidecarContainer), func(t *testing.T) {
+				if tt.inPlacePodLevelResizeEnabled {
+					featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodLevelResourcesVerticalScaling, true)
+				}
```

Comment:
Suggested change
      
    
    
      
          
            
            				if tt.inPlacePodLevelResizeEnabled {
          
          
            
            					featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodLevelResourcesVerticalScaling, true)
          
          
            
            				}
          
          
            
            				featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodLevelResourcesVerticalScaling, tt.inPlacePodLevelResizeEnabled)

I see a few other places actually where the same comment applies, please update them all